### PR TITLE
Review "nodebb" app [#LP-173] [#LP-174]

### PR DESCRIPTION
**Pull request by rosheen33**
connects #LP-173
connects #LP-174

### CODE REVIEW

File: common/djangoapps/nodebb/signals/handlers.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+from common.lib.nodebb_client.client import NodeBBClient
+from xmodule.modulestore.django import modulestore
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from student.models import CourseEnrollment
+
+log = getLogger(__name__)
+
+
+@receiver(post_save, sender=CourseOverview, dispatch_uid="nodebb.signals.handlers.create_category_on_nodebb")
+def create_category_on_nodebb(sender, instance, created, **kwargs):
```
Comment (ziafazal) [inline common/djangoapps/nodebb/signals/handlers.py]: I think this handler should call a celery task and that task should make relevant API calls to nodebb.

File: common/djangoapps/nodebb/signals/handlers.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    if created:
+        status_code, response_body = NodeBBClient().categories.create(name=instance.display_name)
+
+        if status_code != 200:
+            log.error(
+                "Error: Can't create category for the given course {} due to {}".format(instance.id, response_body)
```
Comment (ziafazal) [inline common/djangoapps/nodebb/signals/handlers.py]: `Can't create category` should be `Can't create nodebb cummunity`

File: common/djangoapps/nodebb/signals/handlers.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            )
+
+        log.info('Success: Community created for course {}'.format(instance.id))
```
Comment (ziafazal) [inline common/djangoapps/nodebb/signals/handlers.py]: If status code is not 200 we are still logging a success message.

File: common/djangoapps/nodebb/signals/handlers.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+
+@receiver(post_save, sender=CourseEnrollment, dispatch_uid="nodebb.signals.handlers.join_group_on_nodebb")
+def join_group_on_nodebb(sender, instance, created, **kwargs):
+    if created:
```
Comment (ziafazal) [inline common/djangoapps/nodebb/signals/handlers.py]: I think this handler should call a celery task and that task should make relevant API calls to nodebb

File: common/djangoapps/nodebb/signals/handlers.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+        if status_code != 200:
+            log.error(
+                "Error: Can't create category for the given course {} due to {}".format(instance.id, response_body)
+            )
+
+        log.info('Success: Community created for course {}'.format(instance.id))
+
+
+@receiver(post_save, sender=CourseEnrollment, dispatch_uid="nodebb.signals.handlers.join_group_on_nodebb")
```
Comment (ziafazal) [inline common/djangoapps/nodebb/signals/handlers.py]: I think we should use existing `ENROLL_STATUS_CHANGE`
```
@receiver(ENROLL_STATUS_CHANGE)
def join_group_on_nodebb(sender, event=None, user=None, **kwargs):  # pylint: disable=unused-argument
     if event == EnrollStatusChange.enroll:
```

File: pavelib/prereqs.py [hunk 6]
```diff
@@ -31,6 +31,11 @@
     'requirements/edx/post.txt',
 ]
 
+# PhilU requirements
+PHILU_REQ_FILES = 'requirements/philu/base.txt'
```
Comment (ziafazal) [inline pavelib/prereqs.py]: what about adding a line(`'requirements/philu/base.txt'`)  to `PYTHON_REQ_FILES` list above.

File: common/lib/nodebb_client/categories.py [hunk 7]
```diff
@@ -0,0 +1,8 @@
+from pynodebb.api.categories import Category
+
+
+class PhiluCategory(Category):
```
Comment (ziafazal) [inline common/lib/nodebb_client/categories.py]: can we rename it to`ForumCategory`?

File: common/lib/nodebb_client/users.py [hunk 8]
```diff
@@ -0,0 +1,8 @@
+from pynodebb.api.users import User
+
+
+class PhiluUser(User):
```
Comment (ziafazal) [inline common/lib/nodebb_client/users.py]: Can we rename it to `ForumUser`?

**Comment by zee-pk** [comment]
LGTM, please merge this pull request and create celery task for API calls in an other pull request.

**Pull request merged**